packages/react-composites/src/composites/CallComposite/utils/MediaGalleryUtils.ts

```diff
@@ -0,0 +1,156 @@
+// Licensed under the MIT license.
+
+import { _formatString } from '@internal/acs-ui-common';
+import { RemoteParticipantState } from '@internal/calling-stateful-client';
+import { useEffect, useMemo, useState } from 'react';
+import { useLocale } from '../../localization';
+import { CallAdapterState } from '../adapter';
+import { useSelector } from '../hooks/useSelector';
+
+/**
+ * Custom selector for this hook to retrieve all the participants that are currently
+ * connected to the call.
+ */
+const getRemoteParticipantsConnectedSelector = (callState: CallAdapterState): RemoteParticipantState[] => {
+  return Object.values(callState.call?.remoteParticipants ?? {}).filter(
+    /**
+     * We need to make sure remote participants are in the connected state. If they are not
+     * they might not have their displayName set in the call state just yet.
+     */
+    (p) => p.state === 'Connected'
+  );
+};
+
+type ParticipantChangedAnnouncmentStrings = {
+  participantJoinedNoticeString: string;
+  twoParticipantJoinedNoticeString: string;
+  threeParticipantJoinedNoticeString: string;
+  participantLeftNoticeString: string;
+  twoParticipantLeftNoticeString: string;
+  threeParticipantLeftNoticeString: string;
+  unnamedParticipantChangedString: string;
+  manyParticipantsChanged: string;
+  manyUnnamedParticipantsChanged: string;
+};
+
+/**
+ * sets the announcement string whenever a Participant comes or goes from a call to be
+ * used by the system narrator.
+ *
+ * @returns string to be used by the narrator and Announcer component
+ *
+ * @internal
+ */
+export const useParticipantChangedAnnouncement = (): string => {
+  const locale = useLocale().strings.call;
+  const strings = useMemo(() => {
+    return {
+      participantJoinedNoticeString: locale.participantJoinedNoticeString,
+      twoParticipantJoinedNoticeString: locale.twoParticipantJoinedNoticeString,
+      threeParticipantJoinedNoticeString: locale.threeParticipantJoinedNoticeString,
+      participantLeftNoticeString: locale.participantLeftNoticeString,
+      twoParticipantLeftNoticeString: locale.twoParticipantLeftNoticeString,
+      threeParticipantLeftNoticeString: locale.threeParticipantLeftNoticeString,
+      unnamedParticipantChangedString: locale.unnamedParticipantChangedString,
+      manyParticipantsChanged: locale.manyParticipantsChanged,
+      manyUnnamedParticipantsChanged: locale.manyUnnamedParticipantsChanged
+    };
+  }, [locale]);
+  const [announcerString, setAnnouncerString] = useState<string>('');
+  const remoteParticipants = useSelector(getRemoteParticipantsConnectedSelector);
+  const [previousParticipants, setPreviousParticipants] = useState<RemoteParticipantState[]>(remoteParticipants);
+
+  useEffect(() => {
+    const setParticipantEventString = (string: string): void => {
+      setAnnouncerString('');
+      setAnnouncerString(string);
+    };
+
+    if (previousParticipants.length > remoteParticipants.length) {
```

Comment:
My question about handling the case when folks join and leave at the same time is still open.

We should first compute `whoLeft` and `whoJoined` and only then check for the two cases (`whoLeft.length > 0` and `whoJoined.length > 0`)